homeassistant/components/enocean/sensor.py

```diff
@@ -283,3 +292,21 @@ def value_changed(self, packet):
             self._attr_native_value = "tilt"
 
         self.schedule_update_ha_state()
+
+class EnOceanWindowSensor(EnOceanSensor):
+    """Representation of an EnOcean window sensor device.
+
+    EEPs (EnOcean Equipment Profiles):
+    - D5-00-01 e.g. NodON Reed contact window sensor
+    """
+
+    def value_changed(self, packet):
+        """Update the internal state of the sensor."""
+        action = (packet.data[1] & 0x01)
+
+        if action == 0x01:
+            self._attr_native_value = STATE_CLOSED
+        else:
+            self._attr_native_value = STATE_OPEN
```

Comment:
Why is this not a `binary_sensor`?

Reply:
The documentation describes the binary sensors as "This can typically be one of those batteryless wall switches".
"EnOcean binary sensors have _no state_, they only generate ‘button_pressed’ events."

The new device is not a user actuated wall switch, and it most definitely _HAS a state_. It is essentially the same as the window handle (minus the third state).

Personally I find the naming scheme a bit awkward with binary sensor and sensor, but that's not my call.

Reply:
IMO the documentation is misleading.
Just because the current EnOcean binary sensors do not have states doesn't mean that you can't add binary sensors with a state.

The `1BS/D5-00-01` device should really be a binary sensor, and the documentation adjusted accordingly.

Reply:
The code fits into the "sensors.py" very easily and IMHO is well placed in that file. 

"binary sensors" only serves the one wall switch protocol. It does not have state, nor does it have device classes to differentiate between types.  To keep the code clean, I would rather propose "binary sensors" to be renamed.

Reply:
Just because it's easy doesn't mean it's correct.
You are adding a boolean (Open/Close) sensor, so it should be added to the binary sensor platform.